### PR TITLE
[example/bwmf] Deployable programs.

### PR DESCRIPTION
**Pull request by BaiGang**
This PR includes:

* A matrix proto data generator.
* A deployable main program to start controller/task/data_converter jobs.
* A sample job config.

I'm using this implementation for executing bwmf tasks on a large cluster.

### CODE REVIEW

**Comment by fengjingchao** [comment]
LGTM.

File: example/bwmf/bwmf/main.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	if oErr != nil {
+		log.Fatalf("Failed opening task config file. %s", oErr)
+	}
+	confData, rdErr := ioutil.ReadAll(crd)
+	if rdErr != nil {
+		log.Fatalf("Failed reading task config. %s", rdErr)
+	}
+	log.Printf("conf data: %s", confData)
+	if *jobType == "d" {
+		log.Println("Trying to convert the mat to protobuf and save it to ", *pbResPath)
```
Comment (fengjingchao) [inline example/bwmf/bwmf/main.go]: Put this log into ConvertMatrixTxtToPb()?

File: example/bwmf/bwmf/convert_pb.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/taskgraph/taskgraph/example/bwmf"
+	pb "github.com/taskgraph/taskgraph/example/bwmf/proto"
+	"github.com/taskgraph/taskgraph/filesystem"
+)
+
+func ConvertMatrixTxtToPb(confBytes []byte, path string) {
```
Comment (fengjingchao) [inline example/bwmf/bwmf/convert_pb.go]: Better separate it into two functions? One for convert and one for saving to fs.

File: example/bwmf/bwmf/main.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	jobType := flag.String("job_type", "c", "Job type, either 'c', 't' or 'd'")
+	pbResPath := flag.String("pb_shard_path", "./", "Path to output pb buf. For 'd' job only.")
+	numTasks := flag.Int("num_tasks", 1, "Num of tasks.")
+	numIters := flag.Int("num_iters", 10, "Num of iters for matrix factorization.")
+	latentDim := flag.Int("latent_dim", 100, "Dimensions of latent factors.")
+	taskConfigFile := flag.String("task_config", "", "Path to task config json file.")
+
+	flag.Parse()
+
+	if *jobType != "c" && *jobType != "t" && *jobType != "d" {
```
Comment (fengjingchao) [inline example/bwmf/bwmf/main.go]: I suggest it would be more clear to write the code like:

```
switch jobtype:
  "c":
  "t":
  default: panic
```

**Comment by BaiGang** [comment]
Thanks for the review. @fengjingchao 

I'll be separating bwmf and datautil to two executables.

**Comment by BaiGang** [comment]
Oops. The build failed because of the change in Azure SDK commit 383dafbdf926 [this](https://github.com/Azure/azure-sdk-for-go/commit/383dafbdf92619345cb75e933524504333193f36).

**Comment by plutoshe** [comment]
Sure, I fix it right away.

**Comment by fengjingchao** [comment]
LGTM

**Pull request merged**